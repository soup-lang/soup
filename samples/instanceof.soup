class MyError: Error
    ctor(m) super(m)
        this.message: m

func throwS(s)
    throw Error(s)

func throwM(s)
    throw MyError(s)

try
    throwS('the quick brown fox')
catch e
    console.log('S', e instanceof Error, e instanceof MyError, e.message)

try
    throwM('a lazy dog')
catch e
    console.log('M', e instanceof Error, e instanceof MyError, e.message)

console.log('B', true instanceof Error, false instanceof MyError)
