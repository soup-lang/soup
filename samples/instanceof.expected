S true false the quick brown fox
M true true a lazy dog
B false false
